wisl/lib/ParserAndCompiler/wisl2Gil.ml

```diff
@@ -724,14 +724,30 @@ let rec compile_stmt_list ?(fname = "main") ?(is_loop_prefix = false) stmtl =
         (mk ~stmt_kind:(Multi NotEnd) (), mk ~stmt_kind:(Multi EndNormal) ())
       in
       let v_get = gen_str gvar in
-      let getcmd =
+      let faillab, ctnlab = (gen_str fail_lab, gen_str ctn_lab) in
+      let checkptrcmd =
+        Cmd.GuardedGoto
+          ( Expr.BinOp
+              ( Expr.UnOp (UnOp.TypeOf, comp_e),
+                BinOp.Equal,
+                Expr.Lit (Literal.Type ListType) ),
+            faillab,
+            ctnlab )
+      in
+      let failcmd = Cmd.Fail ("InvalidPointer", []) in
+      let lookupcmd =
         Cmd.LAction (v_get, getcell, [ nth comp_e 0; nth comp_e 1 ])
       in
+      let cmd =
+        [
+          (get_annot, None, checkptrcmd);
+          (getval_annot, Some faillab, failcmd);
```

Comment:
**Q**: What should value of annotation be here?

Reply:
Actually, this might be a question for @NatKarmios now. He knows the wisl lifter better than I do.

@ggwg, annotations are used by the debugger to figure out what GIL code corresponds to what WISL code (or other TL code for other instantations).

Reply:
Yes, I should have said, Gavin and I talked through this last week, including what the annots should be here, though I'm not sure if he's committed that change.

Reply:
Annotation changes have been updated now. Thanks Nat